### PR TITLE
fix(curriculum): fix formatting of forEach method in Todo app project

### LINKED ISSUE #52768: Todo App: Step 17 forEach callback implicit return
### Describe the Issue

I'm not sure I see the point of the `forEach` callback having an implicit return seeing as you can't use the return for anything. I don't know that it helps code readability either.

Current final code:

```
const updateTaskContainer = () => {
  tasksContainer.innerHTML = "";

  taskData.forEach(
    ({ id, title, date, description }) =>
      (tasksContainer.innerHTML += `
      <div class="task" id="${id}">
        <p><strong>Title:</strong> ${title}</p>
        <p><strong>Date:</strong> ${date}</p>
        <p><strong>Description:</strong> ${description}</p>
        <button onclick="editTask(this)" type="button" class="btn">Edit</button>
        <button onclick="deleteTask(this)" type="button" class="btn">Delete</button> 
      </div>
    `)
  );
};
```

With callback function body:

```
const updateTaskContainer = () => {
  tasksContainer.innerHTML = "";

  taskData.forEach(({ id, title, date, description }) => {
    tasksContainer.innerHTML += `
      <div class="task" id="${id}">
        <p><strong>Title:</strong> ${title}</p>
        <p><strong>Date:</strong> ${date}</p>
        <p><strong>Description:</strong> ${description}</p>
        <button onclick="editTask(this)" type="button" class="btn">Edit</button>
        <button onclick="deleteTask(this)" type="button" class="btn">Delete</button> 
      </div>
    `;
  });
};
```

If we want an implicit return, I would prefer a map. The exact formatting is up for debate.

```
const updateTaskContainer = () => {
  tasksContainer.innerHTML = taskData
    .map(
      ({ id, title, date, description }) => `
        <div class="task" id="${id}">
          <p><strong>Title:</strong> ${title}</p>
          <p><strong>Date:</strong> ${date}</p>
          <p><strong>Description:</strong> ${description}</p>
          <button onclick="editTask(this)" type="button" class="btn">Edit</button>
          <button onclick="deleteTask(this)" type="button" class="btn">Delete</button> 
        </div>
      `
    )
    .join("");
};
```

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-localstorage-by-building-a-todo-app/step-17

### Your code

```



```

### Expected behavior

No implicit return or a map. I guess it is just a matter of opinion.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by v-i-k-a-s**
- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52768 

Remove implicit return of arrow function and update by taskData.forEach(()=>{}), done for reg Expressionx in step 17 , update seed code and hint from step17-end #52768
Checklist:

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec94f0de20c086e09b0fc3.md [hunk 1]
```diff
@@ -335,14 +335,16 @@ taskForm.addEventListener("submit", (e) => {
...
+  taskData.forEach(({id, title, date, description}) => {
+      (tasksContainer.innerHTML += `
+        <div class="task" id="${id}">
+    --fcc-editable-region--
+        
+    --fcc-editable-region--
+        </div>
+      `)
+    }
+
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec94f0de20c086e09b0fc3.md:347]: remove extra blank line here

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fac4d1773e7a719b1254de.md [hunk 2]
```diff
@@ -332,16 +332,18 @@ taskForm.addEventListener("submit", (e) => {
...
+        <div class="task" id="${id}">
+          <p><strong>Title:</strong> ${title}</p>
+          <p><strong>Date:</strong> ${date}</p>
+          <p><strong>Description:</strong> ${description}</p>
+          <button type="button" class="btn">Edit</button>
+          <button type="button" class="btn">Delete</button>
+        </div>
+      `)
+    }
+
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fac4d1773e7a719b1254de.md:346]: remove extra blank line here

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64faca774fd9fd74bc084cc9.md [hunk 3]
```diff
@@ -329,16 +329,18 @@ taskForm.addEventListener("submit", (e) => {
...
+        <div class="task" id="${id}">
+          <p><strong>Title:</strong> ${title}</p>
+          <p><strong>Date:</strong> ${date}</p>
+          <p><strong>Description:</strong> ${description}</p>
+          <button type="button" class="btn">Edit</button>
+          <button type="button" class="btn">Delete</button>
+        </div>
+      `)
+    }
+
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64faca774fd9fd74bc084cc9.md:343]: remove extra blank line here

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64facf6180824876f70a2e86.md [hunk 4]
```diff
@@ -345,16 +345,18 @@ taskForm.addEventListener("submit", (e) => {
...
+        <div class="task" id="${id}">
+          <p><strong>Title:</strong> ${title}</p>
+          <p><strong>Date:</strong> ${date}</p>
+          <p><strong>Description:</strong> ${description}</p>
+          <button type="button" class="btn">Edit</button>
+          <button type="button" class="btn">Delete</button>
+        </div>
+      `)
+    }
+
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64facf6180824876f70a2e86.md:359]: remove extra blank line here

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fad07f43a101779cb8692a.md [hunk 5]
```diff
@@ -355,16 +355,18 @@ taskForm.addEventListener("submit", (e) => {
...
+        <div class="task" id="${id}">
+          <p><strong>Title:</strong> ${title}</p>
+          <p><strong>Date:</strong> ${date}</p>
+          <p><strong>Description:</strong> ${description}</p>
+          <button type="button" class="btn">Edit</button>
+          <button type="button" class="btn">Delete</button>
+        </div>
+      `)
+    }
+
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fad07f43a101779cb8692a.md:369]: remove extra blank line here

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fad9cd2eeb1e7ca2ca8c8b.md [hunk 6]
```diff
@@ -341,16 +341,18 @@ taskForm.addEventListener("submit", (e) => {
...
+        <div class="task" id="${id}">
+          <p><strong>Title:</strong> ${title}</p>
+          <p><strong>Date:</strong> ${date}</p>
+          <p><strong>Description:</strong> ${description}</p>
+          <button type="button" class="btn">Edit</button>
+          <button type="button" class="btn">Delete</button>
+        </div>
+      `)
+    }
+
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fad9cd2eeb1e7ca2ca8c8b.md:355]: remove extra blank line here

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64faedcd16a1e985c4c2dc94.md [hunk 7]
```diff
@@ -318,16 +318,18 @@ const updateTaskContainer = () => {
...
+        <div class="task" id="${id}">
+          <p><strong>Title:</strong> ${title}</p>
+          <p><strong>Date:</strong> ${date}</p>
+          <p><strong>Description:</strong> ${description}</p>
+          <button onclick="editTask(this)" type="button" class="btn">Edit</button>
+          <button onclick="deleteTask(this)" type="button" class="btn">Delete</button> 
+        </div>
+      `)
+    }
+
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64faedcd16a1e985c4c2dc94.md:332]: remove extra blank line here

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64faedcd16a1e985c4c2dc94.md (hunk 7 above)
Comment (v-i-k-a-s) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64faedcd16a1e985c4c2dc94.md]: Hi @jdwilkin4 remove all extra line 
Thanks

**Comment by lasjorg** [comment]
I would prefer if we allowed for an implicit return as well, even if we do not ask for it. It is still valid code and someone might write it like that.

Or if we do want to enforce a function body, which I'm also fine with, I think it has to be stated in the written requirements.

**Review by jdwilkin4 — APPROVED** [review]

**Comment by jdwilkin4** [comment]
> I would prefer if we allowed for an implicit return as well, even if we do not ask for it. It is still valid code and someone might write it like that.

Yeah, that's fair. 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md [hunk 8]
```diff
@@ -20,13 +20,13 @@ assert.match(code, /taskData\.forEach\(\s*\(\s*\{\s*(?:id\s*,\s*title\s*,\s*date
 You should use arrow syntax and attach `innerHTML` to `tasksContainer`.
 
 ```js
-assert.match(code, /taskData\.forEach\(\(\{.*\}\)\s*=>\s*(\(\s*tasksContainer\.innerHTML\s*\))?/)
+assert.match(code, /taskData\.forEach\(\(\{.*\}\)\s*=>\s*(\s*\{\(\s*tasksContainer\.innerHTML\s*\)\s*\})?/)
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md:23]: Based on the recent conversation let's update this test to allow for both valid solutions 
https://github.com/freeCodeCamp/freeCodeCamp/pull/52988#issuecomment-1879834476

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md [hunk 9]
```diff
@@ -20,13 +20,13 @@ assert.match(code, /taskData\.forEach\(\s*\(\s*\{\s*(?:id\s*,\s*title\s*,\s*date
...
 ```
 
 You should use addition assignment to set the `innerHTML` of `tasksContainer` to an empty pair of backticks.
 
 ```js
-assert.match(code, /taskData\.forEach\(\(\{.*\}\)\s*=>\s*(\s*\(?tasksContainer\.innerHTML\s*\+=\s*`\s*`\)?)/)
+assert.match(code, /taskData\.forEach\(\(\{.*\}\)\s*=>\s*\{\s*(\s*\(?tasksContainer\.innerHTML\s*\+=\s*`\s*`\)?\s*\})/)
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md:29]: Let's update this test as well to allow for both valid solutions 
https://github.com/freeCodeCamp/freeCodeCamp/pull/52988#issuecomment-1879834476

Comment (v-i-k-a-s) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md]: Make explicit return also possible .
Please let me know if any further changes needed

Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md]: LGTM 👍

**Comment by lasjorg** [comment]
Maybe it's just me but I think the challenge instructions are extremely terse.

> Using arrow syntax, use addition assignment to set the `innerHTML` of `tasksContainer` to empty backticks.

Maybe something like this instead?

> Using arrow syntax complete the `forEach` callback function. Inside the callback function body use an addition assignment to set the `innerHTML` of `tasksContainer` to empty backticks.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md [hunk 10]
```diff
@@ -7,7 +7,7 @@ dashedName: step-17
 
 # --description--
 
-Using arrow syntax, create an implicit return and use addition assignment to set the `innerHTML` of `tasksContainer` to empty backticks.
+Using arrow syntax, use addition assignment to set the `innerHTML` of `tasksContainer` to empty backticks.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md:10]:
```suggestion
Using arrow syntax, complete the `forEach` callback function. Inside the callback function body, use an addition assignment operator to set the `innerHTML` of `tasksContainer` to empty backticks.
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
One last change
Thanks for your hard work on this @v-i-k-a-s 💯

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md (hunk 10 above)
Comment (v-i-k-a-s) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md]: @jdwilkin4 changes the description for step 17

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md [hunk 11]
```diff
@@ -7,7 +7,7 @@ dashedName: step-17
 
 # --description--
 
-Using arrow syntax, create an implicit return and use addition assignment to set the `innerHTML` of `tasksContainer` to empty backticks.
+Using arrow syntax complete the forEach callback function. Inside the callback function body use an addition assignment to set the innerHTML of tasksContainer to empty backticks.
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md:10]:
```suggestion
Using arrow syntax complete the `forEach` callback function. Inside the callback function body use an addition assignment to set the `innerHTML` of `tasksContainer` to empty backticks.
```

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md (hunk 10 above)
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md]: It looks like you didn't include the backticks `` when you implemented this change.
That is what the latest review comment is about 👍

Comment (v-i-k-a-s) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md]: Hi @jdwilkin4 , i missed that things , included in my next commit

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md (hunk 11 above)
Comment (v-i-k-a-s) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ec9282cd547785258cecf2.md]: Hi @lasjorg 
Make the required changes 

Thanks

**Review by lasjorg — APPROVED** [review]
LGTM, thank you for all the extra work.

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**